tests/cxx/std_size_t.cc

```diff
@@ -7,33 +7,22 @@
 //
 //===----------------------------------------------------------------------===//
 
-// This tests a little quirk with the mappings for the new <cstd...> headers.
```

Comment:
I think it's actually better to remove this testcase.

It was a little confusing; I don't think its purpose is to test anything around `size_t`, but rather to demonstrate that mappings depend on include order and use order, with `size_t` as a common example.

Now that `size_t` is more unambiguous, let's get rid of it (I tried to rephrase it in terms of `NULL`/`printf`, but it's not really adding any relevant coverage, IMO)

Reply:
I originally removed it, but then I thought that the file name of the test makes it appropriate for testing that std::size_t  is mapped to <cstddef>.  We don't have any tests for that, do we?

Reply:
We typically don't test mappings, because they're inherently platform specific.

Reply:
> We typically don't test mappings, because they're inherently platform specific.

Makes sense.

BTW, I hadn't found the wording to respond to your request that I keep the ptrdiff comment, but now I have.

> # std::ptrdiff_t is often an architecture specific definition, force the

Typedefs are inherently platform-specific definitions.  If they weren't, they'd use plain `long` (or `int`).  What does that comment add?

Reply:
In the context of IWYU, it implies that `ptrdiff_t` is defined in a private header, possibly multiple different headers. But yeah, I guess that's true for most mapped symbols.

I just don't want to mix that comment removal up with the `size_t` changes, so please just leave it alone for now.

Reply:
I've moved the comment removal to a separate commit.

Reply:
And also removed the test case.